Source/cmake/FindRice.cmake

```
@@ -0,0 +1,139 @@
+# IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR ITS
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+# OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+# WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+# OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+# ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#[=======================================================================[.rst:
+FindRice
+--------------
+
+Find Rice headers and libraries.
+
+Imported Targets
+^^^^^^^^^^^^^^^^
+
+``Rice::Proto``
+  The Rice proto library, if found.
+
+``Rice::Io``
+  The Rice io library, if found.
+
+#]=======================================================================]
+
+find_package(PkgConfig QUIET)
+
+if ("Proto" IN_LIST Rice_FIND_COMPONENTS)
+    pkg_check_modules(PC_RICE_PROTO rice-proto)
+    find_path(RiceProto_INCLUDE_DIR
+        NAMES rice-proto.h
+        HINTS ${PC_RICE_PROTO_INCLUDEDIR}
+        ${PC_RICE_PROTO_INCLUDE_DIRS}
+    )
+
+    find_library(Rice_PROTO_LIBRARY
+        NAMES rice-proto
+        HINTS ${PC_RICE_PROTO_LIBDIR} ${PC_RICE_PROTO_LIBRARY_DIRS}
+    )
+
+    if (Rice_PROTO_LIBRARY)
+        set(RiceProto_FOUND TRUE)
+        if (Rice_FIND_REQUIRED_Proto)
+            list(APPEND Rice_LIBS_FOUND "Proto (required): ${Rice_PROTO_LIBRARY}")
+        else ()
+           list(APPEND Rice_LIBS_FOUND "Proto (optional): ${Rice_PROTO_LIBRARY}")
+        endif ()
+    else ()
+        if (Rice_FIND_REQUIRED_Proto)
+           set(_Rice_REQUIRED_LIBS_FOUND OFF)
+           list(APPEND Rice_LIBS_NOT_FOUND "Proto (required)")
+        else ()
+           list(APPEND Rice_LIBS_NOT_FOUND "Proto (optional)")
+        endif ()
+    endif ()
+endif ()
+
+if ("Io" IN_LIST Rice_FIND_COMPONENTS)
+    pkg_check_modules(PC_RICE_IO rice-io)
+    find_path(RiceIo_INCLUDE_DIR
+        NAMES rice-io.h
+        HINTS ${PC_RICE_IO_INCLUDEDIR}
+        ${PC_RICE_IO_INCLUDE_DIRS}
+    )
+
+    find_library(Rice_IO_LIBRARY
+        NAMES rice-io
+        HINTS ${PC_RICE_IO_LIBDIR} ${PC_RICE_IO_LIBRARY_DIRS}
+    )
+
+    if (Rice_IO_LIBRARY)
+        set(RiceIo_FOUND TRUE)
+        if (Rice_FIND_REQUIRED_Io)
+            list(APPEND Rice_LIBS_FOUND "Io (required): ${Rice_IO_LIBRARY}")
+        else ()
+           list(APPEND Rice_LIBS_FOUND "Io (optional): ${Rice_IO_LIBRARY}")
+        endif ()
+    else ()
+        if (Rice_FIND_REQUIRED_Io)
+           set(_Rice_REQUIRED_LIBS_FOUND OFF)
+           list(APPEND Rice_LIBS_NOT_FOUND "Io (required)")
+        else ()
+           list(APPEND Rice_LIBS_NOT_FOUND "Io (optional)")
+        endif ()
```

Same here, we only need to set `Rice_Io_FOUND`:


```suggestion
        set(Rice_Io_FOUND TRUE)
```